plugins/module_utils/elbv2.py

```diff
@@ -742,289 +730,480 @@ def modify_elb_attributes(self):
+    """This function compares rule_a with rule_b and returns differences
+    between the two rules as a dictionnary
 
-    def add(self):
-        try:
-            # Rules is not a valid parameter for create_listener
-            if "Rules" in self.listener:
-                self.listener.pop("Rules")
-
-            # handle multiple certs by adding only 1 cert during listener creation and make calls to add_listener_certificates to add other certs
-            listener_certificates = self.listener.get("Certificates", [])
-            first_certificate, other_certs = [], []
-            if len(listener_certificates) > 0:
-                first_certificate, other_certs = listener_certificates[0], listener_certificates[1:]
-                self.listener["Certificates"] = [first_certificate]
-            # create listener
-            create_listener_result = AWSRetry.jittered_backoff()(self.connection.create_listener)(
-                LoadBalancerArn=self.elb_arn, **self.listener
-            )
-            # only one cert can be specified per call to add_listener_certificates
-            for cert in other_certs:
-                AWSRetry.jittered_backoff()(self.connection.add_listener_certificates)(
-                    ListenerArn=create_listener_result["Listeners"][0]["ListenerArn"], Certificates=[cert]
-                )
+    Parameters:
+    current_rule: The current listener rule stored in AWS.
+    new_rule: The new listener rule.
 
-        except (BotoCoreError, ClientError) as e:
-            self.module.fail_json_aws(e)
+    Returns:
+    A dictionnary containing the modified attributes.
+    """
 
-    def modify(self):
-        try:
-            # Rules is not a valid parameter for modify_listener
-            if "Rules" in self.listener:
-                self.listener.pop("Rules")
-            AWSRetry.jittered_backoff()(self.connection.modify_listener)(**self.listener)
-        except (BotoCoreError, ClientError) as e:
-            self.module.fail_json_aws(e)
+    modified_rule = {}
 
-    def delete(self):
-        try:
-            AWSRetry.jittered_backoff()(self.connection.delete_listener)(ListenerArn=self.listener)
-        except (BotoCoreError, ClientError) as e:
-            self.module.fail_json_aws(e)
+    # Priority
+    if int(current_rule["Priority"]) != int(new_rule["Priority"]):
+        modified_rule["Priority"] = new_rule["Priority"]
+
+    # Actions
+    actions_match = _compare_rule_actions(current_rule["Actions"], new_rule["Actions"])
+    if not actions_match:
+        modified_rule["Actions"] = new_rule["Actions"]
+
+    # Conditions
+    modified_conditions = [
+        c for c in new_rule["Conditions"] if not _check_rule_condition(current_rule["Conditions"], c)
+    ]
+    if modified_conditions:
+        modified_rule["Conditions"] = modified_conditions
+
+    return modified_rule
+
+
+def _group_rules(
+    current_rules: List[Dict[str, Any]], rules: List[Dict[str, Any]]
```

Comment:
What would you think about splitting this one too?

Reply:
same here, this is a small function